.github/workflows/ci.yml

```
@@ -4,15 +4,31 @@ on:
   pull_request:
     branches: [ main ]
 
+env:
+  RUST_VERSION: "1.65"
+
 jobs:
   build:
 
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
+      with:
+        submodules: recursive
+
+    - name: Set up Rust
+      uses: actions-rs/toolchain@v1
       with:
-        fetch-depth: 0
+        profile: minimal
+        toolchain: ${{ env.RUST_VERSION }}
+        override: true
+
+    - name: Build Rust Project
+      run: cargo build --manifest-path=./backend/rust-bindings/Cargo.toml --release
+    
+    - name: Test Rust Project
+      run: cargo test --manifest-path=./backend/rust-bindings/Cargo.toml --release
```

```suggestion
    - name: Test Rust Project
      run: cargo test --manifest-path=./backend/rust-bindings/Cargo.toml --release

    - name: Clippy Rust Project
      run: cargo clippy --manifest-path=./backend/rust-bindings/Cargo.toml --color=always --tests

    - name: Format Rust Project
      run: cargo fmt --manifest-path=./backend/rust-bindings/Cargo.toml --check
```
I would also add clippy and format checks for the CI.